ui/tests/unit/utils/allocation-stats-tracker-test.js

```diff
@@ -189,13 +189,21 @@ module('Unit | Util | AllocationStatsTracker', function() {
     assert.deepEqual(tracker.get('cpu'), [], 'No tracked cpu yet');
     assert.deepEqual(tracker.get('memory'), [], 'No tracked memory yet');
 
+    const expected = [
+      { task: 'service', reservedCPU: 100, reservedMemory: 256, cpu: [], memory: [] },
+      { task: 'sidecar', reservedCPU: 50, reservedMemory: 128, cpu: [], memory: [] },
+      { task: 'log-shipper', reservedCPU: 50, reservedMemory: 128, cpu: [], memory: [] },
+    ];
+
+    console.log('found');
+    console.log(JSON.stringify(tracker.get('tasks')));
+    console.log('expected');
+    console.log(JSON.stringify(expected));
+    console.log(JSON.stringify(expected) == JSON.stringify(tracker.get('tasks')));
+
     assert.deepEqual(
       tracker.get('tasks'),
-      [
-        { task: 'service', reservedCPU: 100, reservedMemory: 256, cpu: [], memory: [] },
-        { task: 'sidecar', reservedCPU: 50, reservedMemory: 128, cpu: [], memory: [] },
-        { task: 'log-shipper', reservedCPU: 50, reservedMemory: 128, cpu: [], memory: [] },
-      ],
+      expected,
```

Comment:
This test is failing with cryptic message, and I'm unable to tell how the expected value differs from actual value. The json format for both values seem to be the same:

```
not ok 971 Chrome 89.0 - [15 ms] - Unit | Util | AllocationStatsTracker: append appropriately maps a data frame to the tracked stats for cpu and memory for the allocation as well as individual tasks
    ---
        actual: >
            [object Object],[object Object],[object Object]
        expected: >
            [object Object],[object Object],[object Object]
        stack: >
                at Object.<anonymous> (http://localhost:7357/ui/assets/tests.js:21918:14)
                at runTest (http://localhost:7357/ui/assets/test-support.js:27932:36)
                at Test.run (http://localhost:7357/ui/assets/test-support.js:27920:8)
                at http://localhost:7357/ui/assets/test-support.js:28142:15
                at processTaskQueue (http://localhost:7357/ui/assets/test-support.js:27530:32)
                at http://localhost:7357/ui/assets/test-support.js:27534:12
        message: >
            tasks represents the tasks for the allocation with no stats yet
        negative: >
            false
        browser log: |
            LOG: found
            LOG: [{"task":"service","reservedCPU":100,"reservedMemory":256,"cpu":[],"memory":[]},{"task":"sidecar","reservedCPU":50,"reservedMemory":128,"cpu":[],"memory":[]},{"task":"log-shipper","reservedCPU":50,"reservedMemory":128,"cpu":[],"memory":[]}]
            LOG: expected
            LOG: [{"task":"service","reservedCPU":100,"reservedMemory":256,"cpu":[],"memory":[]},{"task":"sidecar","reservedCPU":50,"reservedMemory":128,"cpu":[],"memory":[]},{"task":"log-shipper","reservedCPU":50,"reservedMemory":128,"cpu":[],"memory":[]}]
            LOG: true
```

from: https://app.circleci.com/pipelines/github/hashicorp/nomad/15529/workflows/ea9a55cd-cdcd-4fa0-83e0-995fc61e2adf/jobs/147809/steps

Reply:
Apparently, the issue was that reservedMemoryMax was marked as `undefined` and json marshaller was skipping that field.